elfi/methods/posteriors.py

```diff
@@ -84,36 +84,100 @@ def plot(self, *args, **kwargs):
 
 class BolfiPosterior(Posterior):
 
-    def __init__(self, model, threshold, priors=None, max_opt_iters=10000):
+    def __init__(self, model, threshold=None, priors=None, max_opt_iters=10000):
         super(BolfiPosterior, self).__init__()
         self.threshold = threshold
         self.model = model
         if self.threshold is None:
-            minloc, minval = stochastic_optimization(self.model.eval_mean, self.model.bounds, max_opt_iters)
+            minloc, minval = stochastic_optimization(self.model.predict_mean, self.model.bounds, max_opt_iters)
             self.threshold = minval
             logger.info("Using minimum value of discrepancy estimate mean (%.4f) as threshold" % (self.threshold))
-        self.priors = [None] * model.input_dim
+        self.priors = priors or [None] * model.input_dim
         self.ML, self.ML_val = stochastic_optimization(self._neg_unnormalized_loglikelihood_density, self.model.bounds, max_opt_iters)
         self.MAP, self.MAP_val = stochastic_optimization(self._neg_unnormalized_logposterior_density, self.model.bounds, max_opt_iters)
 
     def logpdf(self, x, norm=False):
+        """
+        Returns the log-posterior pdf at x.
+
+        Note: currently unnormalized.
+
+        Parameters
+        ----------
+        x : np.array
+        norm : boolean
+            Whether to normalize (unsupported).
+
```

Comment:
maybe just remove this? I don't know if there is any general (enough) way to normalize these.